packages/router/useRouteParams/index.ts

```diff
@@ -36,8 +39,15 @@ export function useRouteParams<
     transform,
   } = options
 
-  const transformGet = transform && 'get' in transform ? transform.get : transform ?? ((value: T) => value as any as K)
-  const transformSet = transform && 'set' in transform ? transform.set : (value: K) => value as any as T
+  const transformGet = transform && typeof transform === 'object' && 'get' in transform
```

Comment:
Just an opinion rather than a blocker: IMO we should move ternaries like this into `if` blocks to make the code more readable. Especially since we don't parenthesise any of it 

I suspect there's also a way to fall back to the default once rather than twice

Reply:
I think option 1 seems fine if all works well

there's a small performance gain by having the defaults defined once, so it'd be nice to keep them outside

Reply:
👌done